src/Compilers/Core/Portable/PortableShim.cs

```diff
@@ -504,6 +504,24 @@ internal static class Assembly
                 .CreateDelegate<Func<System.Reflection.Assembly, string, bool, bool, Type>>();
         }
 
+        internal static class AssemblyName
+        {
+            private const string TypeName = "System.Reflection.AssemblyName";
+
+            internal static readonly Type Type = ReflectionUtilities.GetTypeFromEither(
+                contractName: $"{TypeName}, {CoreNames.System_Reflection}",
+                desktopName: TypeName);
+
+            private static readonly PropertyInfo CodeBase = Type
+                .GetTypeInfo()
+                .GetDeclaredProperty(nameof(CodeBase));
```

Comment:
@Pilchie is correct.  The PortableShim type is meant for only APIs that are available in the portable .Net Core subset.  This way when we move to 4.6 then PortableShim can be deleted entirely.